### PR TITLE
chore: migrate from CRA/webpack to Vite

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: #2968 

### Description:
- [x] removes CRA/react-scripts/webpack
- [x] adds Vite with react plugin and frontend proxy
- [x] eslint with vite
- [x] jest
- [x] build
- [x] render deploy
- [x] compare build results

upgraded packages to match versions between monorepos:
- [x] typescript
- [x] prettier/eslint and related packages
- [x] jest
- [x] ts-jest/ts-node

on my m1 air laptop Webpack was taking 3+ minutes for the dev server to start
demo video below where I start a Vite dev server and test live update is <2 minutes
on my laptop Webpack/HMR took ~10 seconds to live update
with Vite - live update is instant

https://user-images.githubusercontent.com/962095/185079445-f9361a70-d074-4e7c-a166-f1127e20bd5f.mp4

### CODE REVIEW

File: packages/frontend/src/setupProxy.js [hunk 1]
```diff
@@ -1,11 +0,0 @@
-const { createProxyMiddleware } = require('http-proxy-middleware');
```
Comment (owlas) [inline packages/frontend/src/setupProxy.js:1]: do we not still need this to have the react proxy in dev? or is this handled by vite?

https://create-react-app.dev/docs/proxying-api-requests-in-development/#configuring-the-proxy-manually

> Note: You do not need to import this file anywhere. It is automatically registered when you start the development server

Comment (owlas) [reply packages/frontend/src/setupProxy.js:1]: nice!

File: packages/common/src/compiler/translator.ts [hunk 2]
```diff
@@ -390,14 +390,16 @@ const modelCanUseMetric = (
...
+
 export const convertExplores = async (
     models: DbtModelNode[],
     loadSources: boolean,
     adapterType: SupportedDbtAdapter,
     metrics: DbtMetric[],
 ): Promise<(Explore | ExploreError)[]> => {
     const tableLineage = translateDbtModelsToTableLineage(models);
-    const [tables, exploreErrors] = models.reduce(
+    const [tables, exploreErrors] = models.reduce<TablesErrorsTuple>(
```
Comment (IrakliJani) [inline packages/common/src/compiler/translator.ts:402]: again, to silence typescript after upgrade

File: packages/frontend/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
+module.exports = {
+    transform: {
+        '\\.jsx?$': 'esbuild-jest',
+        '\\.tsx?$': 'esbuild-jest',
+        '\\.css$': 'esbuild-jest',
```
Comment (IrakliJani) [inline packages/frontend/jest.config.js:6]: using esbuild-jest to transform js files

**Comment by IrakliJani** [comment]
@lightdash/engineering
how to test locally:
- pull down
- yarn
- run development environment

**Comment by rephus** [comment]
Updating the frontend is pretty fast (most of the time) 
But sometimes fully refreshing the page takes up to 1 minute ! to load 

Peek 2022-08-24 09-16

Could we also stop clearing the console when refreshing happens ? 
It is a good start, but let's test this thoroughly to ensure it is stable and better than what we have.

**Comment by IrakliJani** [comment]
@rephus

> But sometimes fully refreshing the page takes up to 1 minute ! to load
yep, its annoying. I'll try to optimise that.

> Updating the frontend is pretty fast (most of the time)
can you provide an example when it's slow to HMR?

> Could we also stop clearing the console when refreshing happens ?
yeah, clearing should have been disabled, probably this is not the right config https://github.com/lightdash/lightdash/pull/2969/files#diff-9a6976d7abd91fcbcf9be8127b4b866f3f693fed188e2d31765b0020ec78b600R32

> It is a good start, but let's test this thoroughly to ensure it is stable and better than what we have.
💯

**Comment by IrakliJani** [comment]
@ZeRego have not had a chance to wrap this up yet
still needs to implement dependency pre-building to increase some initial load time in dev mode: https://vitejs.dev/guide/dep-pre-bundling.html

**Comment by ZeRego** [comment]
- `yarn dev` is fast. But I assume that is because it not bundling anything. 
```
  VITE v3.0.9  ready in 2417 ms

  ➜  Local:   http://localhost:3000/
  ➜  Network: http://172.22.0.3:3000/
```
- It takes quite a long time to load the page since is loading a lot of files.

Screenshot 2022-09-07 at 11 14 20

- Hot reload is fast 👍 
- Page refresh seems fast for me 👍 
- Don't know why but I had to disable the `Optimize imports` option when saving a file since it was removing all the imports :/ Not a big deal.

Sep-07-2022 11-18-24

File: packages/backend/src/clients/EmailClient/EmailClient.ts [hunk 4]
```diff
@@ -94,7 +94,7 @@ export default class EmailClient {
             try {
                 const info = await this.transporter.sendMail(options);
                 Logger.debug(`Email sent: ${info.messageId}`);
-            } catch (error) {
+            } catch (error: any) {
```
[the remark this answers is not in the document]
Comment (ZeRego) [reply packages/backend/src/clients/EmailClient/EmailClient.ts:97]: We can change the tsconfig instead and improving the error typing in a separate PR
```
  "compilerOptions": {
    // ... other stuff
    "useUnknownInCatchVariables": false
  }
```

**Comment by IrakliJani** [comment]
@ZeRego yeah, initial load timer is a fake advertisement. it compiles files on the fly. I'm gonna spend some time today to finish it up.

**Pull request closed without merging**